native/kotlin/api/kotlin/src/main/kotlin/rs/wordpress/api/kotlin/WpRequestExecutor.kt

```diff
@@ -36,15 +51,25 @@ class WpRequestExecutor(
                 }
             }
 
-            okHttpClient.newCall(requestBuilder.build()).execute().use { response ->
-                return@withContext WpNetworkResponse(
-                    body = response.body?.bytes() ?: ByteArray(0),
-                    statusCode = response.code.toUShort(),
-                    responseHeaderMap = WpNetworkHeaderMap.fromMultiMap(response.headers.toMultimap()),
-                    requestUrl = request.url(),
-                    requestHeaderMap = request.headerMap()
-                )
+            val urlRequest = requestBuilder.build()
+
+            try {
+                getClient().newCall(urlRequest).execute().use { response ->
+                    return@withContext WpNetworkResponse(
+                        body = response.body?.bytes() ?: ByteArray(0),
+                        statusCode = response.code.toUShort(),
+                        responseHeaderMap = WpNetworkHeaderMap.fromMultiMap(response.headers.toMultimap()),
+                        requestUrl = request.url(),
+                        requestHeaderMap = request.headerMap()
+                    )
+                }
+            } catch (e: SSLPeerUnverifiedException) {
+                handleInvalidSSLError(e, urlRequest)
+            } catch (e: UnknownHostException) {
+                handleUnknownHostException(e, urlRequest)
             }
+
+            error("Unknown Error occurred")
```

Comment:
This shouldn't be necessary and if I am not mistaken, the program execution can never make it here. If `try` succeeds it'll return the value, if the exception is caught it'll throw another exception and if it's not caught it'll stop the execution.

Compiler doesn't seem to be very smart about this as removing this line results in an error. However, if you change the helper functions to return the exception instead of throwing it and throw inside the catch blocks, then it'll stop complaining. In my opinion, this is also the correct way to do this and I was going to make the suggestion regardless of this line.